flocker/node/agents/test/test_blockdevice.py

```diff
@@ -148,19 +148,43 @@
     blockdevice_id=_create_blockdevice_id_for_test(dataset.dataset_id),
 ))
 
+# Text generation is slow, in order to speed up tests and make the output more
+# readable, use short strings and a small legible alphabet. Given the way
+# metadata is used in the code this should not be detrimental to test coverage.
+_METADATA_STRATEGY = text(average_size=3, min_size=1, alphabet="CGAT")
+
+DESIRED_DATASET_ATTRIBUTE_STRATEGIES = {
+    'dataset_id': uuids(),
+    'maximum_size': integers(min_value=0).map(
+        lambda n:
+        LOOPBACK_MINIMUM_ALLOCATABLE_SIZE
+        + n*LOOPBACK_ALLOCATION_UNIT),
+    'metadata': dictionaries(keys=_METADATA_STRATEGY,
+                             values=_METADATA_STRATEGY),
+    'mount_point': builds(FilePath, sampled_from([
+        '/flocker/abc', '/flocker/xyz',
+    ])),
+}
+
 DESIRED_DATASET_STRATEGY = tagged_union_strategy(
     DesiredDataset,
-    {
-        'dataset_id': uuids(),
-        'maximum_size': integers(min_value=0).map(
-            lambda n:
-            LOOPBACK_MINIMUM_ALLOCATABLE_SIZE
-            + n*LOOPBACK_ALLOCATION_UNIT),
-        'metadata': dictionaries(keys=text(), values=text()),
-        'mount_point': builds(FilePath, sampled_from([
-            '/flocker/abc', '/flocker/xyz',
-        ])),
-    }
+    DESIRED_DATASET_ATTRIBUTE_STRATEGIES
+)
+
+_NoSuchThing = object()
+
+# This strategy creates two `DESIRED_DATASET_STRATEGY`s with as much in common
+# as possible. This is supposed to approximate two potentially different
+# `DesiredDataset` states for the same dataset.
+TWO_DESIRED_DATASET_STRATEGY = DESIRED_DATASET_STRATEGY.flatmap(
+    lambda x: tuples(just(x), tagged_union_strategy(
```

Comment:
Yea I tried to write it in a way that could be re-used. My general rule is if you only want to use something once, do it in-line when you are tempted to copy it make it a function.

I'd like to see a second use case to see where the right abstraction layer is before making this a proper utility in common.algebraic or something. If something strikes me in the next hour I'll code it up, otherwise I'm thinking about leaving this as is for now.